### PR TITLE
Testbench initialization

### PR DESCRIPTION
**Pull request by eselznick2**
**SFR Edit-** Moving summary for changes from email into this PR (2026-04-14)

## Change Summary 

`ClockDivider2X.vhd`
- Major change: renamed entity to ClockDivider2XPorts / ClockDivider2X, and changed phase split (eg div is 10, it was high for 5,6,7,8,9,0 and low 1,2,3,4)

`PatternFinder.vhd`
- Major change: cleanup ('00' -> "00")

`PeekRingBuffer.vhd`
- Major change: compile/syntax and assignment fixes (duplicate PopAddress declaration removed, missing parens fixed, signal assignment operator fixes for LastHeaderEnd/LastFooterEnd).
- Question: TB shows current full/empty/count behavior effectively gives 2047 usable bytes in a 2048 address (shows full after 2047 writes). If this is intended then fine, idk if full 2048 needed.
- Question: empty, full, and count do calculations using U/X before they can be reset and this gives assertion warnings, sim still runs / assuming this setup won't mess up syn.

`RtcCounter.vhd`
- Major change: GeneratedPPS moved into registered process behavior, sensitivity cleanup, and reset handling tightened (SetChangedTime/GeneratedPPS reset behavior made explicit).

`UartRxFifoExtClk.vhd`
- Major change: switched FIFO instance from gated_fifo interface to fifo interface (wone_i/rone_i to we_i/re_i) with wiring cleanup.

`UartRxFifoExtClkPeek.vhd`
- Major change: updated internal PeekRingBuffer component declaration/map with LastHeaderEnd, LastFooterEnd, PayloadLen, HeaderFooterPayloadLenMatches, currently tied to internal Unused* signals.

`UartRxFifoParity.vhd`
- Major change: If ParityErr goes high, then no write to fifo

`UartRxParity.vhd`
- Major change: receive logic update to more closely match UartRxRaw logic (PARITY_EVEN generic, parity accumulation/check, stop-bit/parity handling, ParityErr output behavior).

`UartTxParity.vhd`
- Major change: Go edge is now accepted only when Busy = 0 (prevents retrigger while active).

`VariableClockDivider2X.vhd`
- Major change: renamed entity/arch to VariableClockDivider2XPorts / VariableClockDivider2X and adjusted phase logic around terminal_count handling (same as non variable)

`VariableOneShot.vhd`
- Major change: rm /2 out of bounds.

`crc_fifo.vhd`
- Major change: Added generic reset value to crc value (default is xFFFFFFFF).

`fifo_gen.vhd`
- Major change: read/write gate logic rewritten to avoid U/X, count output now combinational from state (can prob just be written straight from counter_r), full/empty boundary equations adjusted (empty/full etc... based on incoming read/write) so update on next edge, read output reset behavior (if we don't need empty/full/count logic immediatley on next I can revert this logic)

`fifo_peek.vhd`
- Major change: Peek from RAM, reset-safe read outputs, same empty/full/count logic as fifo_gen.vhd

`gated_fifo.vhd`
- Major change: N/A
- Question: Write is not tied to data_i so you could request a write, let it reach the fifo, and change data_i to write wrong data. Should write data be registered with a write?

`gated_fifo_peek.vhd`
- Major change: aligned with new fifo_peek
- Question: same data_i question as gated_fifo.vhd

`CGraphTypes.vhd`
- Major change: N/A

`ClockDivider.vhd`
- Major change: N/A

`crc_byte.vhd`
- Major change: N/A

`crc_stream.vhd`
- Major change: N/A

`FieldLatcher.vhd`
- Major change: N/A

`IBufP1.vhd`
- Major change: N/A

`IBufP2.vhd`
- Major change: N/A
- Question: This is used in a lot of different UartFifo implementations. But because the buf has no reset value, if there is a high value within the buffer when a reset hits, this high value will still propogate and can cause some wrong post-reset logic (like writing zero immediately after reset). If we're only expecting rst at initialization, this is fine, but if we're expecting some mid-run rst then it might be worth to add a generic rst value to buffers.

`IBufP3.vhd`
- Major change: N/A

`OneShot.vhd`
- Major change: N/A

`PeekRam.vhd`
- Major change: N/A

`PPSCount.vhd`
- Major change: N/A

`UartRx.vhd`
- Major change: N/A

`UartRxExtClk.vhd`
- Major change: N/A

`UartRxRaw.vhd`
- Major change: N/A

`UartTx.vhd`
- Major change: N/A

`UartTxFifo.vhd`
- Major change: change component name for sim

`UartTxFifoExtClk.vhd`
- Major change: N/A

`UartTxFifoParity.vhd`
- Major change: N/A

`VariableClockDivider.vhd`
- Major change: N/A

# Testbench Overview

## Clock/Timing Dividers

`ClockDivider_tb.vhd`
- Reset polarity and initial state checks across configs.
- Confirms expected divide cadence (for example /10 and /6 behavior).
- Also tested mid-run resets from both output phases.

`ClockDivider2X_tb.vhd`
- Same core checks as `ClockDivider_tb.vhd`, but for the 2X path.
- Focus here is the updated phase split behavior plus reset recovery.

`VariableClockDivider_tb.vhd`
- Mainly to make sure programmable terminal count still behaves like the fixed divider.
- Retune-at-runtime cases are covered (no reset in between).
- Includes terminal-count edge cases.

`VariableClockDivider2X_tb.vhd`
- Same idea as `VariableClockDivider_tb.vhd`, applied to the 2X implementation.
- Checks dynamic terminal-count changes and reset recovery.

## One-Shot Pulsing

`OneShot_tb.vhd`
- Reset behavior plus trigger/pretrigger timing checks.
- Exercises delay variants and one-shot latch/retrigger boundaries.
- Includes reset-before-trigger and reset-after-trigger paths.

`VariableOneShot_tb.vhd`
- Programmable delay values (small/normal/large).
- Covers delay changes before trigger, pretrigger variants, and reset recovery.

## UART RX / RX FIFO

`UartRx_tb.vhd`
- Basic decode, back-to-back bytes, and byte-range traffic.
- Reset-during-reception and break-condition handling.
- Checks completion signaling and output hold behavior.

`UartRxRaw_tb.vhd`
- Make sure sampling/enable gating and stop-bit timing rules hold.
- Basic decode, full byte range, completion signals (normal behavior)
- Includes baud mismatch, back-to-back frames, and reset mid-frame.

`UartRxExtClk_tb.vhd`
- Same coverage style as `UartRx_tb.vhd`, but with external clocking.
- Also checks ext-clock-specific stop-bit timing thresholds.

`UartRxParity_tb.vhd`
- Parity mode matrix (odd/even), parity accumulation, and parity error signaling.
- Stop-bit timing and baud mismatch cases.
- Reset during active frame is covered too.

`UartRxFifoExtClk_tb.vhd`
- This one is mostly about FIFO integration after the interface swap.
- Single/burst/overlap traffic, read-while-write behavior, and drain ordering.
- Full/empty boundaries plus resets with buffered data and active frames.

`UartRxFifoExtClkPeek_tb.vhd`
- Adds peek + multipop coverage on top of RX FIFO behavior.
- Pop-to-empty/refill/full-to-empty sequences are included.
- Also checks simultaneous pop+write and skip/address edge cases.

`UartRxFifoParity_tb.vhd`
- Main check here: bad parity frames do not get written into FIFO.
- Includes good/bad mixed streams, read timing overlap, boundaries, and reset recovery.

## UART TX / TX FIFO

`UartTx_tb.vhd`
- Reset/idle and normal TX data transmission.
- Go/Busy interaction (no accidental retrigger while busy).
- Reset during transmission is covered.

`UartTxParity_tb.vhd`
- Same core TX checks as `UartTx_tb.vhd`, with parity framing enabled.
- Mainly to confirm Busy-gated `Go` behavior stays solid with parity enabled.
- Sending correct Parity bit under heavy load.

`UartTxFifo_tb.vhd`
- Covers FIFO fed TX flow from empty to queued multi-byte sends.
- Verifies CTS blocking/unblocking behavior and handoff timing.
- Includes full-queue boundaries and reset during queued transmission.

`UartTxFifoExtClk_tb.vhd`
- Covers FIFO fed TX flow from empty to queued multi-byte sends under external clk.
- Verifies CTS blocking/unblocking behavior and handoff timing.
- Includes full-queue boundaries and reset during queued/active transmission.

`UartTxFifoParity_tb.vhd`
- Parity FIFO TX flow under single/queued sends.
- CTS gating, queue boundaries, and reset during active transmission are all covered.

## FIFO / Peek

`fifo_gen_tb.vhd`
- Core FIFO baseline: reset, normal read/write, burst traffic, and boundaries.
- Simultaneous read/write is exercised across empty, steady-state, near-full, and full.
- `r_ack` timing and reset behavior during active traffic are checked.

`fifo_peek_tb.vhd`
- Adds peek + multipop coverage on top of basic FIFO behavior.
- Tests wraparound (still a bit fishy), skip logic, and mixed read/write traffic.
- Includes overlap edge cases (empty, near full, full).
- Also re-checks behavior after resets.

`gated_fifo_tb.vhd`
- Same FIFO boundary/read-write checks, but with edge-triggered strobes.
- Includes held-high strobe behavior and `r_ack` handling.

`gated_fifo_peek_tb.vhd`
- `gated_fifo_tb` coverage plus peek/multipop behavior.
- Mainly to make sure gated strobes do not break skip/peek paths.

`PeekRam_tb.vhd`
- Straightforward RAM checks: write/read/overwrite/same-cycle interactions.
- No-write-when-disabled and reset-after-write behavior included.

`PeekRingBuffer_tb.vhd`
- This one is mostly about correctness around tricky ring-buffer edges.
- Reset defaults, wrap, full/empty transitions, and pop advancement are all exercised.
- Header/footer marker and metadata updates are explicitly checked (not fully done yet).
- TB also exposes the current one-slot-empty behavior (2047 usable entries at depth 2048).

## CRC / Pattern / Latching / Counters

`crc_byte_tb.vhd`
- Quick check that byte-wise CRC combinational logic matches reference vectors.

`crc_stream_tb.vhd`
- Stream CRC checks over reset, zero stream, and simple text payload cases.

`crc_fifo_tb.vhd`
- Deeper coverage here since this block has more moving parts.
- CRC over varied patterns (zeros, ones, PRNG, address slices).
- Full-range sweep plus back-to-back runs without reset.
- Reset handling is tested, including reset during active CRC computation.

`PatternFinder_tb.vhd` (in progress)
- Steps through match/mismatch cases for the pattern detector.
- Makes sure it recovers correctly after a bad sequence.
- Also checks reset behavior mid-stream.

`FieldLatcher_tb.vhd`
- Byte capture and shift sequencing checks across multi-byte fields.
- Also covers held write-request behavior and reset timing cases.

`RtcCounter_tb.vhd`
- Free-run ms/sec progression and rollover checks.
- PPS sync/grace timing, SetTime handling, and generated PPS behavior.

`PPSCount_tb.vhd`
- Simple reset + nominal PPS counting checks.

## Small TypeTBs

`CGraphTypes_tb.vhd`
- Quick type/package checks (so the CGraphTypes.vhd can be built).

`IBufP1_tb.vhd`, `IBufP2_tb.vhd`, `IBufP3_tb.vhd`
- Simple sanity checks that values propagate through the buffers correctly.

Full Catalog:
Each test for each tb and what exactly it is testing
[testbench_test_catalog.md](https://github.com/user-attachments/files/26946737/testbench_test_catalog.md)

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: FPGASim
+
+on: [push]
```
Comment (stefi07) [inline .github/workflows/build-and-test.yml]: The workflow only triggers on push, which means that testbench results won't appear in the PR checks panel. It would make the review process smoother for a reviewer to have easy access to test result information. Consider adding pull_request as a trigger:
```suggestion
on: [push, pull_request]
```

File: include/fpga/testbenches/waves/ClockDivider.gtkw [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[*]
+[*] GTKWave Analyzer v3.3.104 (w)1999-2020 BSI
+[*] Sun Mar  8 07:24:51 2026
+[*]
+[dumpfile] "/home/ellis/ant/PearlTesting/include/fpga/testbenches/work/ClockDivider_tb.ghw"
```
Comment (stefi07) [inline include/fpga/testbenches/waves/ClockDivider.gtkw]: All .gtkw files contain hardcoded absolute paths (`/home/ellis/ant/firmware/.`..) for `[dumpfile]` and `[savefile]`. I don’t think these are used by the CI simulation run, so please consider:
- Should this directory be added to `.gitignore` instead of version-controlled to avoid noisy diffs from local regeneration?
- If they need to be versioned, replace absolute paths with relative paths.

File: include/fpga/ClockDivider2X.vhd [hunk 3]
```diff
@@ -27,7 +27,7 @@ use IEEE.STD_LOGIC_1164.ALL;
 use IEEE.STD_LOGIC_UNSIGNED.ALL;
 use IEEE.NUMERIC_STD.all;
 
-entity ClockDividerPorts is
+entity ClockDivider2XPorts is
```
Comment (stefi07) [inline include/fpga/ClockDivider2X.vhd:30]: Renaming `ClockDividerPorts` to `ClockDivider2XPorts` reduces ambiguity, which is great. However, it’s a breaking change for any code that references the old name, so please confirm all references were updated before merging; I don’t see corresponding updates outside of `include/fpga` right now.

File: include/fpga/testbenches/crc_byte.vhd
Comment (stefi07) [inline include/fpga/testbenches/crc_byte.vhd:1]: Is this a first draft of `crc_byte_tb.vhd`? The filename is missing the `_tb` suffix, so the `Makefile` won't pick it up and it never runs. Since `crc_byte_tb.vhd` already exists, please delete this file to avoid confusion.

File: include/fpga/UartRxParity.vhd [hunk 4]
```diff
@@ -26,77 +26,100 @@ library ieee;
    use ieee.std_logic_1164.all;
    
 entity UartRxParity is
+  generic (
+      PARITY_EVEN : natural := 0 --1=odd, 0=even
```
Comment (stefi07) [inline include/fpga/UartRxParity.vhd:30]: I find the `PARITY_EVEN` naming confusing: it implies `1 = even` parity, but the comment says the opposite. I understand this follows `UartTxParity.vhd`, but is that a standard convention that anybody modifying the codebase would be familiar with?

If it isn’t standard, consider renaming both to `PARITY_ODD` (such that setting the value to 1 enables odd parity). Leaving the current name risks someone toggling the wrong parity when they change the code.

File: include/fpga/UartRxFifoParity.vhd
Comment (stefi07) [inline include/fpga/UartRxFifoParity.vhd:1]: `UartRxParity` now exposes a `PARITY_EVEN` option, but it isn’t propagated, so parity is silently fixed to the default (`UartTxParity` does expose its `PARITY_EVEN`). Is that intentional?

If the intent is to always use the default parity, please add a comment here stating that for clarity.

If it should be configurable, please expose the `PARITY_EVEN` option.

File: include/fpga/VariableClockDivider2X.vhd [hunk 5]
```diff
@@ -27,7 +27,7 @@ use IEEE.STD_LOGIC_1164.ALL;
 use IEEE.STD_LOGIC_UNSIGNED.ALL;
 use IEEE.NUMERIC_STD.all;
 
-entity VariableClockDividerPorts is
+entity VariableClockDivider2XPorts is
```
Comment (stefi07) [inline include/fpga/VariableClockDivider2X.vhd:30]: Same comment as for ClockDivider2X - it’s a breaking change for any code that references the old name, so please confirm all references were updated before merging; I don’t see corresponding updates outside of include/fpga right now.

**Review by stefi07 — COMMENTED** [review]
The PR description and `.md` summary are useful. Please also include that information closer to the code:
- Future commit messages would benefit from more detail so the intent of each change is clear in `git log`/`blame`.
- Please add short summary at the top of each testbench describing what it tests. That information currently exists in a `.md` file attached to the description of this PR, but the tests’ purpose should travel with the code rather than a separate file that can get out of sync more easily. The repo uses `doxygen`, so you can just add:

```
/**
  * \brief This testbench tests the following
  *
  */
```

File: include/fpga/testbenches/crc_byte.vhd
Comment (eselznick2) [reply include/fpga/testbenches/crc_byte.vhd:1]: Yes, this was an earlier draft I misplaced, I'll remove it

File: include/fpga/testbenches/waves/ClockDivider.gtkw (hunk 2 above)
Comment (eselznick2) [reply include/fpga/testbenches/waves/ClockDivider.gtkw]: That's scary, this got passed me. I'll change these to be relative.

**Comment by eselznick2** [comment]
> The PR description and `.md` summary are useful. Please also include that information closer to the code:
> 
> * Future commit messages would benefit from more detail so the intent of each change is clear in `git log`/`blame`.
> * Please add short summary at the top of each testbench describing what it tests. That information currently exists in a `.md` file attached to the description of this PR, but the tests’ purpose should travel with the code rather than a separate file that can get out of sync more easily. The repo uses `doxygen`, so you can just add:
> 
> ```
> /**
>   * \brief This testbench tests the following
>   *
>   */
> ```



> The PR description and `.md` summary are useful. Please also include that information closer to the code:
> 
> * Future commit messages would benefit from more detail so the intent of each change is clear in `git log`/`blame`.
> * Please add short summary at the top of each testbench describing what it tests. That information currently exists in a `.md` file attached to the description of this PR, but the tests’ purpose should travel with the code rather than a separate file that can get out of sync more easily. The repo uses `doxygen`, so you can just add:
> 
> ```
> /**
>   * \brief This testbench tests the following
>   *
>   */
> ```

I added doxygen commenting on all tb files. Let me know if this is a good amount of info or if you'd prefer more/less

**Comment by eselznick2** [comment]
## Summary

Follow-up fixes and coverage improvements requested or found during review. This branch, in addition to previous changes now introduces reset behavior in the `IBufP*` blocks already under test, standardizes several divider-related testbenches, updates `PPSCount` edge-detection behavior and tests, refactors UART RX FIFO parity handling, adds baud-skew characterization coverage across the UART RX family, brings in additional SPI-family fixes/testbenches, `PhaseComparator` signed-delta fixes, `CGraphDmDacRam*` read-after-write fixes, and testbench reporting improvements.

## Related Issues

Related issue: #8

## Changes

- Added reset generic/default handling to the `IBufP*` modules that already have testbench coverage, and updated the affected RTL/tests to match
- Added Doxygen headers/comments across all FPGA testbenches
- Updated `PPSCount` behavior so `PPSDetected` asserts after 2 edges instead of 1, and updated the testbench to reflect the new behavior
- Synchronized the `ClockDivider*` and `OneShot*` and `variable*` testbenches so they follow a more consistent structure
- Added mid-run changing-`tc` edge cases and longer stress tests to the clock divider / one-shot testbenches
- Added quick testbenches for `IOBufP1`, `IOBufP2`, and `IOBufP3`
- Refactored `UartRxFifoParity` to better match the other UART RX FIFO blocks, including separate master clock and 16x sample clock handling
- Updated `UartRx` and `UartRxParity` so start-bit detection requires a falling edge and re-samples the start bit at sample count 8/16 to avoid start glitches
- Added baud-skew sweep tests across the UART RX family to find supported baud tolerance ranges (tested with 115200 Bd but can work with other rates if we want to test with those). Baud range is reported at end of log e.g. `Baud range of 115200 Bd: 109232 - 121969 baud (-5.549864035191298% / +5.4632451496476895% tolerance)`. 
- In general most UartRx have a skew tolerance of ~5.55%.
- Updated the FPGA testbench `Makefile` so `make all` runs testbenches in parallel for faster CI
- Running make all will extract baud rates from UartRx modules logs and will be placed in doxygen comment (all UartRx*_tb.vhd will automatically have the found baud skew tolerance in their doxygen comments)
-  Added SPI-family testbench coverage for `SpiMaster*`, `SpiDevice*`, `SpiDac*`, and `SpiExtBus`, plus GTKWave files for `SpiDevice` and `SpiMaster`
- Fixed SPI transfer timing/data-latching behavior so MOSI data remains stable through active transfers and completion signaling is more robust
- Added testbench coverage for `CGraphDmDacRam`, `CGraphDmDacRamFlat`, `CGraphDmTypes`, and `PhaseComparator`
- Fixed `CGraphDmDacRam*` read-after-write behavior during simultaneous access cases
- Corrected `PhaseComparator` signed-delta handling and saturation behavior, and added reset support
- Expanded `tb_utils_pkg.vhd` with additional helpers and clearer PASS reporting

## Type of Change

- [x] Bug fix
- [ ] New feature
- [x] Tests change / update
- [x] Refactor / code cleanup
- [x] FPGA / HDL change
- [ ] Firmware / embedded change
- [x] Build / CI change
- [x] Documentation update

## Testing

**FPGA simulation:** Ran `make all` in `include/fpga/testbenches`  
**Hardware test:** Not run  
**Unit tests:** No C++ unit tests affected

## Checklist

- [x] Branch is based on `master` (and rebased on the latest upstream, if forked)
- [x] All FPGA simulation tests pass (`make all` in `include/fpga/testbenches`)
- [ ] C++ code is formatted with `clang-format`
- [x] New functionality is covered by tests
- [x] Doxygen header and comments added/updated
- [x] No new compiler warnings introduced
- [x] No merge conflicts with the target branch
- [x] `CONTRIBUTING.md` guidelines followed
- [x] [Pull Request Checklist](https://uasal.github.io/uasal_development_guide/git/pull_request_checklist.html) guidelines followed

## Breaking Changes

`IBufP*` modules now have rst signal and a generic RST_VALUE. I've changed all modules currently with testbenches to work with this change.

`PhaseComparator` now treats `Delta` as a signed quantity with explicit signed saturation behavior.

## Additional Notes

This PR update includes changes made since the last review of the larger testbench-initialization branch, rather than re-listing the original full set of testbench additions. Additional testbenches from `testbench-initialization-batch2` have been folded into this PR.

File: include/fpga/testbenches/update_doxygen_baud.sh
Comment (stefi07) [inline include/fpga/testbenches/update_doxygen_baud.sh:1]: As far as I can tell, this updates a subset of testbenches with a baud range derived from running the testbench. I appreciate the completeness in documenting this, but it rings alarms for me in terms of change tracking, since running the testbenches would result in uncommitted changes, which a user would find either confusing or commit without necessarily reflecting the correct ranges. I’d remove this bash and instead replace the '--! Latest baud result: ' doxygen lines with a brief note that once the testbench is run, the baud range can be found in the respective log file, <log_file_name>.

File: include/fpga/UartRxFifoParity.vhd
Comment (stefi07) [inline include/fpga/UartRxFifoParity.vhd:1]: I think this is the only one from my old comments that wasn’t addressed. It's okay if there is a reason to not take any action over it, but a quick comment on why would be useful.

`UartRXParity` introduces a `PARITY_EVEN` generic, but that isn’t propagated here, so parity is silently fixed to the default. Is that intentional? If the intent is to always use the default parity, please add a comment here (and everywhere else where that’s the case) stating that for clarity. If it should be configurable, please expose the `PARITY_EVEN` option.

File: include/fpga/testbenches/Makefile [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+	@for f in $(RTL_FILES) $(TB_FILES:%=$(TB_DIR)/%); do \
+		[ -f $$f ] && ghdl -a $(GHDL_FLAGS) $$f; \
+	done
+	@for tb in $(TB_NAMES); do \
+		ghdl -e $(GHDL_FLAGS) $${tb}_tb; \
+	done
+
+rebuild: clean build
+
+define run_tb
```
Comment (stefi07) [inline include/fpga/testbenches/Makefile:44]:
```suggestion
# run_tb must always exit 0 such that every testbench runs. The actual pass/fail gate is the log grep in all.
define run_tb
```

Small documentation point. Running the testbenches never actually fails, which is great and allows for all testbenches to run, rather than aborting after the first failure, but it's non-obvious on a first read and looks like a bug (a FAIL print that doesn't fail the target). A one-line comment above run_tb along the lines of the suggested one would help with readability.

File: include/fpga/testbenches/tb_utils_pkg.vhd
Comment (stefi07) [inline include/fpga/testbenches/tb_utils_pkg.vhd:1]: The utility functions files has grown significantly. It could be worth considering to split it in 2-3 utility files grouped per concern for future maintainability. If you don't foresee it expanding further perhaps it's unnecessary, but if future work will require more helper functions, it will be easier to split it now than later.

Comment (eselznick2) [reply include/fpga/testbenches/tb_utils_pkg.vhd:1]: Yes. This is a good point, I do see it continuing to grow, I'll likely seperate all Uart functions into a Uart helper file, same with SPI, and then other generics.

File: include/fpga/UartRxFifoParity.vhd
Comment (eselznick2) [reply include/fpga/UartRxFifoParity.vhd:1]: I can check with Summer. This may cause some backward compatability problems, but if not then I can go ahead and switch it around so its clearer. Either way I'll make the documentation clearer as well.
